### ISSUE TITLE
Misuse of Terminology (GROSS and NET)

### ISSUE THREAD

**Issue by menshakov** [vaened/php-price-engine#1]
### Issue: Misuse of Terminology (GROSS and NET)

Hi! Thanks for providing such a useful library. I hope to integrate it into my projects in the future.

However, I noticed an issue with the terminology used in the library: **the terms "GROSS" and "NET" appear to be swapped in the method names.**  But all finally calculations are good.

To clarify:
- **GROSS price** is the price **including all taxes**.
- **NET price** is the price **excluding taxes**.

Here’s an example that illustrates the problem:

```php
$cashier = new SimpleCashier(
    Amount::taxable(
        Money::of(100, 'USD'),
        TaxCodes::any()
    ),
    quantity: 1,
    taxes: Taxes::from([
        Tax\Inclusive::proporcional(21, 'IVA'),
    ])
);

echo "unitPrice GROSS: {$cashier->unitPrice()->gross()}" . PHP_EOL; // unitPrice GROSS: USD 82.64
echo "unitPrice NET: {$cashier->unitPrice()->net()}" . PHP_EOL; // unitPrice NET: USD 99.99
echo "subtotal: {$cashier->subtotal()->gross()}" . PHP_EOL; // subtotal: USD 82.64
echo "taxes: {$cashier->taxes()->total()}" . PHP_EOL; // taxes: USD 17.35
```

Based on the output:
- The **unitPrice()->gross()** method actually returns the **NET price** (`USD 82.64`).
- The **unitPrice()->net()** method actually returns the **GROSS price** (`USD 99.99`).

---

Additionally, I noticed the following sections of the code may be affected:

1. In the `total()` method:
   ```php
   public function total(): Money
   {
       return $this->subtotal()
                   ->gross()
                   ->plus($this->taxes()->total())
                   ->plus($this->charges()->total())
                   ->minus($this->discounts()->total());
   }
   ```
   - Here, it seems the `->gross()` is used where the **NET subtotal** should be.

2. In the `initializePrice()` method:
   ```php
   protected function initializePrice(Money $grossUnitPrice, Adjusters $taxes): void
   {
       $this->price = new Price(
           $grossUnitPrice,
           netUnitPrice: $grossUnitPrice->plus(
               AdjustmentManager::totalize($grossUnitPrice, $taxes)
           ),
       );
   }
   ```
   - The `netUnitPrice` is calculated by **adding taxes** to the `grossUnitPrice`, which is contrary to the expected behavior.

---

Could you please confirm if this is indeed a mix-up? If so, adjusting the terminology would greatly improve clarity and prevent potential confusion for developers.

Thanks again for your hard work on this library! 😊

**Comment by vaened** [comment]
You’re absolutely right! I mixed up the GROSS and NET terms. Just fixed it, so everything should be correct now. Thanks for catching that! 😄

Also, sorry for the late reply—I either missed the GitHub notification or didn’t get one. Appreciate your patience!

**Comment by menshakov** [comment]
Thank you for the library. It's great!
